llm-service/app/ai/indexing/readers/base_reader.py

```diff
@@ -1,21 +1,48 @@
+import os
+import tempfile
 from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
 from pathlib import Path
-from typing import List
+from typing import List, Optional, Set
 
+from detect_secrets.core.secrets_collection import SecretsCollection
+from detect_secrets.settings import default_settings
 from llama_index.core.node_parser import SentenceSplitter
 from llama_index.core.schema import Document, TextNode
+from presidio_analyzer import AnalyzerEngine
+from presidio_anonymizer import AnonymizerEngine
+
+
+@dataclass
+class ReaderConfig:
+    block_secrets: bool = False
+    anonymize_pii: bool = False
+
+
+@dataclass
+class ChunksResult:
+    chunks: List[TextNode] = field(default_factory=list)
+    # If present, the chunks contained secrets and the chunks should be empty
+    secret_types: Optional[Set[str]] = None
+    # If true, the chunks contained PII and the chunks contain anonymized text
+    pii_found: bool = False
 
 
 class BaseReader(ABC):
     def __init__(
-        self, splitter: SentenceSplitter, document_id: str, data_source_id: int
+        self,
+        splitter: SentenceSplitter,
+        document_id: str,
+        data_source_id: int,
+        config: ReaderConfig = ReaderConfig(),
```

Comment:
Not a good idea to assign an object as an argument's default value, unless we set

```python
@dataclass(frozen=True)
class ReaderConfig
```

This `ReaderConfig` instance will be shared by all `BaseReader` instances. If any of them mutate their `self.config`, it will affect all other readers.

Reply:
Oh, nice!

Reply:
The better practice from what I understand is

```python
    config: Optional[ReaderConfig] = None,
):
    ...
    if self.config is None:
        self.config = ReaderConfig()
```

Reply:
For future readers, this is a very common gotcha in Python. The

```python
config: ReaderConfig = ReaderConfig(),
```

is executed when the `BaseReader` class is defined, so the `ReaderConfig()` instantiation happens once, gets stored with the `BaseReader` class definition, and the resulting `ReaderConfig` instance gets propagated to all `BaseReader` instances.

```python
>>> from dataclasses import dataclass
>>> 
>>> @dataclass
... class Config:
...     block_secrets: bool = True
... 
>>> class Reader:
...     def __init__(self, config=Config()):
...         self.config = config
...     def stop_blocking_secrets(self):
...         self.config.block_secrets = False
... 
>>> r1 = Reader()
>>> r2 = Reader()
>>> 
>>> r1.config.block_secrets
True
>>> r2.config.block_secrets
True
>>> 
>>> r1.stop_blocking_secrets()
>>> r1.config.block_secrets
False
>>> r2.config.block_secrets
False
```

Reply:
Did the optional path